application/client/src/app/ui/views/toolbar/search/input/recent.ts

```diff
@@ -1,99 +1,45 @@
-import { SafeHtml } from '@angular/platform-browser';
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
 import { FormControl } from '@angular/forms';
-import { wrapMatchesToSafeHtml } from '@ui/env/globals';
+import { Matcher } from '@wasm/matcher/pkg/matcher';
 import { Entry, EntryConvertable } from '@platform/types/storage/entry';
 import { error } from '@platform/env/logger';
 import { Storage } from '@env/fsstorage';
 import { SetupLogger, LoggerInterface } from '@platform/entity/logger';
 
-import * as obj from '@platform/env/obj';
-
-export class Recent implements EntryConvertable {
-    public value: string = '';
-    public filter: string = '';
-    public used: number = 0;
-
-    private _valueLowerCase: string = '';
-    private _filterLowerCase: string = '';
-
-    constructor(value: string) {
-        this.value = value;
-        this._valueLowerCase = value.toLowerCase();
-    }
-
-    public html(): SafeHtml {
-        return wrapMatchesToSafeHtml(this.filter, this.value);
-    }
-
-    public setFilter(filter: string) {
-        this.filter = filter;
-        this._filterLowerCase = filter.toLowerCase();
-    }
-
-    public filtered(): boolean {
-        return this._valueLowerCase.indexOf(this._filterLowerCase) !== -1;
-    }
-
-    public entry(): {
-        to(): Entry;
-        from(entry: Entry): Error | undefined;
-        hash(): string;
-        uuid(): string;
-        updated(): undefined;
-    } {
-        return {
-            to: (): Entry => {
-                return {
-                    uuid: this.value,
-                    content: JSON.stringify({
-                        filter: this.value,
-                        used: this.used,
-                    }),
-                };
-            },
-            from: (entry: Entry): Error | undefined => {
-                try {
-                    const def: {
-                        filter: string;
-                        used: number;
-                    } = JSON.parse(entry.content);
-                    this.value = obj.getAsNotEmptyString(def, 'filter');
-                    this.used = obj.getAsValidNumber(def, 'used');
-                    this._valueLowerCase = this.value.toLowerCase();
-                } catch (e) {
-                    return new Error(error(e));
-                }
-                return undefined;
-            },
-            hash: (): string => {
-                return `${this.value}/${this.used}`;
-            },
-            uuid: (): string => {
-                return this.value;
-            },
-            updated: (): undefined => {
-                return undefined;
-            },
-        };
-    }
+interface IRecent {
+    value: string;
+    html_value: SafeHtml;
```

Comment:
I believe with removing of `Recent` class you break of the process of correct saving recent filters on the backend.
```class Recent implements EntryConvertable {...}```
inheriting from `EntryConvertable` makes it convertible to format, which is used to save data on the backend.
In general your changes should not touch exist functionality